### ISSUE TITLE
False negative with `Style/AccessorGrouping`

### ISSUE THREAD

**Issue by mlwyatt** [rubocop/rubocop#11891]
When `attr_accessor` is preceeded by a call to Rails 7's `encrypts` method (or any macro method), `Style/AccessorGrouping` produces false negatives. 

--------

## Expected behavior

The `attr_accessor` line below should trigger this cop.
```
app/models/user.rb:6:3: C: [Correctable] Style/AccessorGrouping: Use one attribute per attr_accessor.
  attr_accessor :one, :two
```

## Actual behavior

Rubocop does not complain at all.

## Steps to reproduce the problem

`app/models/user.rb`
```ruby
# frozen_string_literal: true

class User < ApplicationRecord
  encrypts :token

  attr_accessor :one, :two
end
```

`.rubocop.yml`
```yml
Style/AccessorGrouping:
  Enabled: true
  EnforcedStyle: separated
```

I tested replacing `encrypts` with `validates`, replacing `encrypts` with `this_method_does_not_exist`, and changing `Style/AccessorGrouping` back to the default of `EnforcedStyle: grouped` and with every test, this cop was silent when the `attr_accessors` were grouped and separated.

I also tested `grouped` with a 3 separate lines (adding `attr_accessor :three`) and it did complain about `attr_accessor :two` and `attr_accessor :three` needing to be grouped, but ignored `attr_accessor :one`

## RuboCop version

```
$ [bundle exec] rubocop -V
1.49.0 (using Parser 3.2.2.1, rubocop-ast 1.28.1, running on ruby 3.1.2) [aarch64-linux]
  - rubocop-performance 1.16.0
  - rubocop-rails 2.19.1
  - rubocop-rubomatic 1.0.0
```
